packages/block-editor/src/components/list-view/block.js

```diff
@@ -274,6 +284,28 @@ function ListViewBlock( {
 				</>
 			) }
 
+			{ isLocked && (
+				<TreeGridCell
+					className="block-editor-list-view-block__lock-cell"
+					aria-selected={ !! isSelected }
+				>
+					{ ( { ref, tabIndex, onFocus } ) => (
+						<Button
+							disabled
+							ref={ ref }
+							tabIndex={ tabIndex }
+							onFocus={ onFocus }
+							icon={ lock }
+							label={ sprintf(
+								/* translators: %s: block name */
+								__( 'Locked %s' ),
+								blockInformation.title
+							) }
+						/>
+					) }
+				</TreeGridCell>
+			) }
```

Comment:
I think it can be simpler than making it a button in a cell, the cells should only really be used for focusable elements. An alternative would be to add it to the ListViewBlockSelectButton component as an `Icon`, after the anchor? That component's children are laid out using flexbox, so hopefully the icon can be made to justify/align itself at the end.

The other part that I think should be tackled is announcing the locked status to screenreaders.

I was having a look to see how the selected block is announced so that this could be done in the same way, but I can't see it in the code. There might have been a regression.

Reply:
Moving status icon into `ListViewBlockSelectButton` makes sense.

> The other part that I think should be tackled is announcing the locked status to screenreaders.

You're right. Currently, locked blocks aren't announced. I am adding this issue to my to-do list. Thanks for checking and catching the issue, @talldan.